### ISSUE TITLE
Regression in immediate sign

### ISSUE THREAD

**Issue by radare** [aquynh/capstone#362]
fails r2's test `t/cmd_rop`

```
-  0x000000b5     80b801000000b9  cmp byte [rax + 1], 0xb9
+  0x000000b5     80b801000000b9  cmp byte [rax + 1], -0x47
```

**Comment by radare** [comment]
This is the commit that causes the regression: 21b9b25e9dae4af0ef309d4089a54e53b8f5b479

**Comment by aquynh** [comment]
That commit is for pcrel instructions, thus cannot cause this problem.

**Comment by aquynh** [comment]
hmm what branch did you use? the "next" branch still produces output `cmp byte ptr [rax + 1], 0xb9`.

(i havent fixed this for 3.0.3 release)

**Comment by radare** [comment]
Confirmed, works fine in latest next.
